### ISSUE TITLE
Without camera permission for android expo 5.1

### ISSUE THREAD

**Issue by brandonklk** [expo/expo#13073]
### Summary

when downloading the app he asks me for permission but when i use the scanner to open the camera he doesn't have permission.

app expo do not open camera for open app

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

40.0.1

### Environment

Expo CLI 4.4.3 environment info:
    System:
      OS: Windows 10 10.0.19042
    Binaries:
      Node: 14.16.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.10 - C:\Users\myuser\AppData\Roaming\npm\yarn.CMD
      npm: 7.11.2 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ~40.0.0 => 40.0.1
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: https://github.com/expo/react-native/archive/sdk-41.0.0.tar.gz => 0.63.2
      react-native-web: ~0.13.12 => 0.13.18
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

download the expo app from the store and try to open the camera to scan qr code

**Comment by c-mcdonald** [comment]
I have just run into this issue on Android 5.1.

Previously on SDK 41, the Camera.requestPermissionAsync() resolved the status to "granted". Now on SDK 42 and latest builds of the Expo Go app, Camera.requestPermissionAsync() always resolves to "denied" despite the fact it has been granted by the app.

I have verified this on both an Expo SDK 42 managed and bare workflow builds of my app, and also even in the Expo Go App. If You open the Expo Go app on Android 5.1 and try to use the camera to scan a QR code, it will pop a dialog saying "Permission  needed", despite the settings app saying the Camera permission has been granted.

I have also checked whether using the now deprecated expo-permissions package resolves the issue, however it does not.
